src/coreclr/gc/gc.cpp

```diff
@@ -18218,6 +18243,14 @@ int gc_heap::generation_to_condemn (int n_initial,
         }
     }
 
+#ifdef USE_REGIONS
+    if (!try_get_new_free_region())
```

Comment:
I don't understand why we need a new free region here - could you explain?

Reply:
it's to prepare for possibly needing to get a new region for gen0 in the later phases.